File: CommandLineInput.cpp
// Command Line Input in C++


#include <iostream>
#include <cstdlib> 
#include <string>
using namespace std;


int main(int argc, char* argv[]) 
{
  cout << "Number of argument, argc = " << argc << endl;
  cout << "  " << endl; 

  for(int i = 0; i < argc; i++)
    cout << "argv[" << i << "] = " << argv[i] << endl;
  
  
  cout << "  " << endl; 
  for(int i = 1; i < argc; i++)
    cout << atoi(argv[i]) << endl; 
  
  return 0;
} 

File: CommandLineInput.py
# Command Line Input in Python 

# To run the code, type in the shell: 
# python CommandLineInput.py arg1 arg2 333 333.0
# Note that arg1 and arg2 are two dummy variables



# import sys module 
import sys

# Get the total number of args passed to the demo.py
totargs  = len(sys.argv)
 
# Get the arguments list 
listargs = str(sys.argv)


print '                 '
print 'The Number of arguments passed: %d' %totargs
print 'Argument List: %s' %listargs
print '                 '


# Pharsing args one by one 
print ("Script name:     %s" % str(sys.argv[0]))
print ("First argument:  %s" % str(sys.argv[1]))
print ("Second argument: %s" % str(sys.argv[2]))
print ("Third argument:  %d" % int(sys.argv[3]))
print ("Fourth argument: %f" % float(sys.argv[4]))
